Пересобранный на webpack проект, с улучшенным фильтром по поиску товаров

AlexeyMoseev opened the pull request.

AlexeyMoseev committed:
1)Пересобрал проект с помощью webpack
2)Переделал функцию поиска товаров, реализовав функцию debounce

Roman-Maksimov commented on index.html:
```diff
@@ -9,15 +9,14 @@
 		<header class="header">
 			<form class="search-field">
-				<button class="search-button">
+				<button disabled class="search-button">
```
используйте полную форму атрибута `disabled="disabled"`

Roman-Maksimov commented on src/js/listApi.js:
```diff
@@ -0,0 +1,49 @@
+
+listApi()
+
+export default function listApi() {
```
Технически, разницы нет, но обычно делают так:
```javascript
function listApi() {}

listApi();

export default listApi;
```
Так читабельнее

Roman-Maksimov commented on src/js/listApi.js:
```diff
@@ -0,0 +1,49 @@
+
+const urlList = 'http://localhost:3006/item'
+
+listApi()
```
С точки зрения архитектуры не желательно, чтобы модуль выполнял функцию контроллера? В данном случае, модуль предоставляет некую функцию, но модуль не должен ничего знать и беспокоиться о самом вызове этой функции. Это должно происходить где-то снаружи, там, где вы реализуете основную бизнес-логику приложения.

Roman-Maksimov reviewed (changes requested):
+ Комменты к предыдущему PR

AlexeyMoseev committed:
1. Исправил все замечания, которые были даны еще в предыдущей ветке
2. Использовал полную форму атрибута disabled="disabled" вместо сокращенной
3. Сделал небольшие правки для лучшей читабельности кода и улучщающие общую архитектуру приложения
